### DISCUSSION TITLE
Authentication not working when using a basePath

### DISCUSSION THREAD

**Discussion by jezza1245** [nextauthjs/next-auth#11321] [Help]
Hello folks! I am currently using next-auth@5.0.0-beta.19 on next 14.1.3 and am having an issue with the auth callback action no longer working after deploying my site on a basePath (All was working fine without a base path and I have added the basePath property to my config). I am wondering if anyone has experienced similar or has any tips for things I could try; as I've followed the documentation as far as I could but it doesn't seem to be recognising the callback action now?

**Current walkthrough of the issue:**
1. Click login button calls the signIn method exported from my NextAuth initialisation where the basePath property is set to include the location of my [...nextauth] handlers including the subpath ```community/api/auth```
2. The authorization endoint for my B2C application /oauth2/v2.0/authorize is successful and returns to the configured redirect URI at http://localhost:3000/community/api/auth/callback/azureb2c 
3. I then see the issue 
`[auth][error] UnknownAction: Cannot parse action at /api/auth/callback/azureb2c.`

The application seems to not be handling this correctly but I'm unsure where the real issue lies; as the redirect uri provided in my b2c application is correct 'http://localhost:3000/community/api/auth/callback/azureb2c' (includes the basepath), but the error I am seeing seems to suggest it's receiving and trying to invoke an action at /api/auth/callback/azureb2c which does not contain the subpath but is 'internally' correct for the application.

I have some middleware where I have console logged the incoming request and the result is as follows:
```
export async function middleware(request: NextRequest) {
  console.log('->', request)
  return NextResponse.next()
}
```
returns:
```
{
    nextUrl: {
        href: 'http://localhost:3000/community/api/auth/callback/azureb2c?code=<code>,
        ...
        pathname: '/api/auth/callback/azureb2c',
    },
    ...,
    url: 'http://localhost:3000/community/api/auth/callback/azureb2c?code=<code>',
    ...
}
``` 

**Project setup:**
My auth api is setup at the following directory `app/api/auth/[...nextauth]`
Both my NextJS config and NextAuth config have both had the relevant basePath properties added like so:

**next.config.mjs**
```
const nextConfig = {
  output: 'standalone',
  basePath:'/community',
  ...
}

```

**src/auth/auth.ts**
```
export const {
  handlers: { GET, POST },
  auth,
  signIn, // The used signIn method
  signOut,
} = NextAuth({
  providers: [MyAzureB2CProvider],
  ...,
  trustHost: true,
  basePath: '/community/api/auth',
  debug: true,
})
```

I would really appreciate any help if there is something I'm missing or could try; else I will raise an issue for this.

Thanks :) 
Jez

**Comment by nktnet1** [comment]
There's an open pull request summarising the issue here:
- https://github.com/nextauthjs/next-auth/pull/10797

More discussions here:
- https://github.com/nextauthjs/next-auth/discussions/9274
